platform/state/platform_state.proto

```diff
@@ -0,0 +1,373 @@
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.state;
+
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.state.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.platform.state">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * PlatformState MUST represent the current consensus state of the platform entity.
+ * The state SHOULD contain the following components:
+ */
+message PlatformState {
+    /**
+     * The address book for this round.
+     *
+     * The address book MUST represent the latest consensus view of node
+     * addresses.
+     */
+    AddressBook address_book = 1;
+
+    /**
+     * The previous address book.
+     *
+     * A temporary workaround until dynamic address books are supported.
+     * If present, the previous address book SHOULD reflect the address
+     * book from the preceding round.
+     */
+    AddressBook previous_address_book = 2;
+
+    /**
+     * The round of this state.
+     *
+     * MUST represent the handling of all transactions reaching
+     * consensus in all previous rounds. <br>
+     * The first state (genesis state) SHALL have a round of 0, because
+     * the first round is defined as round 1.
+     */
+    uint64 round = 3;
+
+    /**
+     * The running event hash.
+     *
+     * Computed by the consensus event stream. <br>
+     * SHOULD be deleted once the consensus event stream is retired.
+     */
+    bytes legacy_running_event_hash = 4;
+
+    /**
+     * The consensus timestamp for this signed state.
+     *
+     * MUST represent the consensus of the first transaction in this round.
+     */
+    proto.Timestamp consensus_timestamp = 5;
+
+    /**
+     * The version of the application software.
+     *
+     * MUST accurately reflect the software version that created this state.
+     */
+    SoftwareVersion creation_software_version = 6;
+
+    /**
+     * The epoch hash of this state.
+     *
+     * MUST be updated whenever emergency recovery is performed.
+     */
+    bytes epoch_hash = 7;
+
+    /**
+     * The next epoch hash.
+     *
+     * Used to update the epoch hash at the next round boundary. <br>
+     * SHOULD NOT be part of the hash and MUST NOT be serialized.
```

Comment:
Everything in the protobuf will be serialized, it's basically unavoidable.
So should the epoch hash values be stored separately (not in _network_ state at all)?

Reply:
In fact, it should be removed entirely. Cody mentioned here - https://www.notion.so/swirldslabs/Proposed-Platform-State-Structure-d37ae2c018f94b81ac884745ca4ee022 - that, as we no longer support emergency reconnect, these fields are no longer useful and should be removed.